### PR TITLE
fix: field `time` in `Header` and `TransactionResponse`

### PR DESCRIPTION
**Pull request by cheng404**
Hello, I found out a bug about the `time` field in `client.query` results.

the `time` value of the following code is always `None` 

### 1. transaction response
```
let tx_id = "0x7e46c2447d3788be0a41550edaebed99fe6b28c60d1c01186fc0d70d72c38a68";
let provider = Provider::connect("node-beta-2.fuel.network").await.unwrap();
let tx = provider.get_transaction_by_id(tx_id).await.unwrap();
println!("{:#?}", tx);
```

### 2. block header
```
let block_id = "0xba4f5c3197f72f752b82cc46c170b550f5ac6e320656683b2693af0b748d5dd4";
let provider = Provider::connect("node-beta-2.fuel.network").await.unwrap();
let block = provider.block(block_id).await.unwrap();
println!("{:#?}", block);
```

### Fix

use `to_unix()` to convert `TAI64` to timestamp

### CODE REVIEW

File: packages/fuels-types/src/block.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ pub struct Header {
 
 impl From<ClientHeader> for Header {
     fn from(client_header: ClientHeader) -> Self {
-        let naive = NaiveDateTime::from_timestamp_opt(client_header.time.0 .0 as i64, 0);
+        let naive = NaiveDateTime::from_timestamp_opt(client_header.time.0 .to_unix(), 0);
```
Comment (MujkicA) [inline packages/fuels-types/src/block.rs:21]:
```suggestion
        let naive = NaiveDateTime::from_timestamp_opt(client_header.time.0.to_unix(), 0);
```

**Review by MujkicA — DISMISSED** [review]
Good catch. Thank you for contributing!
Left a small formatting fix for the CI to pass.

**Review by iqdecay — CHANGES_REQUESTED** [review]
Thank you! This is good, but it's better if you can add a test that shows that the bug you experienced before is no longer here. Lmk if you need help writing the test :smile:

**Comment by cheng404** [comment]
time parameter of `produce_blocks` should be `TAI64 ` format
https://github.com/FuelLabs/fuels-rs/blob/fce0631cf8778a04d70cf8fdffa0d207b211c4b9/packages/fuels-signers/src/provider.rs#L414

I noticed that there are some changes in `fuel-core` structure of project, so adding 'tai64' dependence is a kind of temporary solution and I will follow up on the changes.

**Comment by digorithm** [comment]
@cheng404 Do you mind syncing this branch with master?

**Comment by cheng404** [comment]
> 
Done. :)

**Review by MujkicA — APPROVED** [review]
LGTM!
Once again, thanks for contributing.

**Review by iqdecay — APPROVED** [review]
nice, thanks for your contribution! @cheng404 I'm letting you the pleasure of clicking the green "Merge" button :smile:

**Comment by cheng404** [comment]
> nice, thanks for your contribution! @cheng404 I'm letting you the pleasure of clicking the green "Merge" button 😄
It's my pleasure. I have no `write access` to this repository, and can't merge and close PR

**Comment by iqdecay** [comment]
> It's my pleasure. I have no `write access` to this repository, and can't merge and close PR
Woops I didn't know that haha. Merged! :tada:

**Pull request merged**